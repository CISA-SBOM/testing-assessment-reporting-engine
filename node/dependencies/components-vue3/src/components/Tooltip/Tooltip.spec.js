import { describe, expect, it } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import Component from "./Tooltip.vue";

describe("Tooltip.vue", () => {
  it("is a Vue instance", () => {
    const wrapper = shallowMount(Component);
    expect(wrapper.vm).toBeTruthy();
  });
  it("matches snapshot", () => {
    const wrapper = shallowMount(Component);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("matches snapshot with no props assigned", () => {
    const propsData = {};
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("matches snapshot with prop assigned", () => {
    const propsData = {
      placement: 'bottom',
      size: 'sm',
      tooltipClass: 'test',
      triggerClass: 'test'
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

});
