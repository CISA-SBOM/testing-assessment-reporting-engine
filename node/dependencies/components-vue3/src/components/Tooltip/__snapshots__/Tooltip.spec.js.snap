// Vitest Snapshot v1

exports[`Tooltip.vue > matches snapshot 1`] = `
<div
  class="inline-block"
  data-id="sds-tooltip"
>
  <div
    class="inline-block w-full"
  >
    
    <div>
      
      
    </div>
    
  </div>
  <!---->
</div>
`;

exports[`Tooltip.vue > matches snapshot with no props assigned 1`] = `
"<div class=\\"inline-block\\" data-id=\\"sds-tooltip\\">
  <div class=\\"inline-block w-full\\">
    <div></div>
  </div>
  <!---->
</div>"
`;

exports[`Tooltip.vue > matches snapshot with prop assigned 1`] = `
"<div class=\\"inline-block\\" data-id=\\"sds-tooltip\\" tooltipclass=\\"test\\" triggerclass=\\"test\\">
  <div class=\\"inline-block w-full\\">
    <div></div>
  </div>
  <!---->
</div>"
`;
